tests/deploys.go

```diff
@@ -0,0 +1,83 @@
+package tests
+
+import (
+	"testing"
+	"time"
+
+	"github.com/nspcc-dev/neo-go/pkg/neotest"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/stretchr/testify/require"
+)
+
+type (
+	// RegisterContractInNNSParams is a param struct for RegisterContractInNNS func.
+	RegisterContractInNNSParams struct {
+		// Name is a contract name. Name must not be a top level domain. For instance: netmap.
+		// The final registered domain name will be concatenated with `.neofs`.
+		Name         string
+		ContractHash util.Uint160
+		Email        string
+		Refresh      int64
+		Retry        int64
+		// Expire in milliseconds
+		Expire int64
+		TTL    int64
+	}
+)
+
+// NewRegisterContractInNNSParams is a constructor for RegisterContractInNNSParams.
+func NewRegisterContractInNNSParams(name string, contractHash util.Uint160) RegisterContractInNNSParams {
+	return RegisterContractInNNSParams{
+		Name:         name,
+		ContractHash: contractHash,
+		Email:        "ops@nspcc.ru",
+		Refresh:      3600,
+		Retry:        600,
+		Expire:       int64(365 * 24 * time.Hour / time.Millisecond), // 1 year.
+		TTL:          3600,
+	}
+}
+
+func nnsContractInvoker(t *testing.T, e *neotest.Executor) *neotest.ContractInvoker {
+	nnsHash, err := e.Chain.GetContractScriptHash(1)
+	require.NoError(t, err)
+
+	return e.CommitteeInvoker(nnsHash)
+}
+
+func getContractHash(t *testing.T, e *neotest.Executor, name string) util.Uint160 {
+	nnsInv := nnsContractInvoker(t, e)
+	resolveResult, err := nnsInv.TestInvoke(t, "resolve", name, 16)
+	require.NoError(t, err)
+
+	contractHashBytes, err := resolveResult.Pop().Array()[0].TryBytes()
+	require.NoError(t, err)
+
+	netMapContractHash, err := util.Uint160DecodeStringLE(string(contractHashBytes))
+	require.NoError(t, err)
+
+	return netMapContractHash
+}
+
+// RegisterContractInNNS make registration of the contract hash in nns contract.
+func RegisterContractInNNS(t *testing.T, e *neotest.Executor, params RegisterContractInNNSParams) {
+	nnsHash, err := e.Chain.GetContractScriptHash(1)
+	require.NoError(t, err)
+
+	nnsInv := e.CommitteeInvoker(nnsHash)
+
+	nnsInv.Invoke(t, true, "register", params.Name+".neofs", e.CommitteeHash, params.Email, params.Refresh, params.Retry, params.Email, params.TTL)
+	nnsInv.Invoke(t, nil, "addRecord", params.Name+".neofs", 16, params.ContractHash.StringLE())
```

Comment:
can we use https://github.com/nspcc-dev/neofs-contract/blob/db6dbb9fa0116f95b7532c84266c4ca84cb94227/contracts/nns/recordtype/recordtype.go#L15?

Reply:
Updated